lib/assert.js

```diff
@@ -185,6 +185,50 @@ function isArguments(object) {
   return Object.prototype.toString.call(object) == '[object Arguments]';
 }
 
+function compareMap(a, b, strict) {
+  // check size of the maps
+  if (a.size !== b.size)
+    return false;
+
+  if (!compareMapKeys(a, b, strict))
+    return false;
+
+  return compareMapKeys(b, a, strict);
+}
+
+function compareMapKeys(a, b, strict) {
+  // iterate over all keys and check if they exist in the other
+  for (let kvs of a) {
+    var key = kvs[0];
+    var val = kvs[1];
+    if (!b.has(key)) {
+      return false;
+    }
+
+    if (!_deepEqual(val, b.get(key), strict))
+      return false;
+  }
+  return true;
+}
+
+function compareSet(a, b, strict) {
+  if (a.size !== b.size)
+    return false;
+
+  if (!compareSetItems(a, b, strict))
+    return false;
+
+  return compareSetItems(b, a, strict);
+}
+
+function compareSetItems(a, b, strict) {
+  for (let item of a) {
+    if (!b.has(item))
```

Comment:
I think I agree with @Fishrock123 here. Order should probably be checked as well IMO.